compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIESplitLogicalObjFifos.cpp

```diff
@@ -121,6 +121,45 @@ FailureOr<int64_t> getSplitStride(ArrayRef<AMDAIE::DmaCpyNdOp> dmaOps,
   return splitStride;
 }
 
+/// Fetch total no. of unique pairs of L2<->L1 Copy ops. This would helps us
+/// figure out the split factor for all LogicalObjectFifos. Basically we get to
+/// decide how many splits to perform for a particular L2 ObjectFifo based on
+/// the total unique L2<->L1 Copy ops.
+/// Eg:
+///      %lhs = LOF_on_L2
+///      %a = LOF_on_L1_0
+///      %b = LOF_on_L1_1
+///      %c = LOF_on_L1_2
+///      DMA(%a, %lhs)
+///      DMA(%b, %lhs)
+///      DMA(%c, %lhs)
+///      DMA(%b, %lhs)
+///      DMA(%c, %lhs)
+///
+///    In the above snippet although we have 5 DMA ops for L2<->L1, only 3 of
+///    them are unique. Hence we'd split %lhs into 3 unique splits, instead
+///    of 5.
+static FailureOr<int64_t> fetchTotalUniqueL2L1(
+    SmallVector<CopyOpInterface> copyLikeOps, bool fetchTarget) {
```

Comment:
You could use `template <CopyOpOperateOn OperateOn>` instead of `bool fetchTarget` to make this a compile-time check.